### PR TITLE
Use named fields in Build and Version structs

### PR DESCRIPTION
**Pull request by zeenix**
Best use explicit names to clarity purposes.

### CODE REVIEW

**Comment by zeenix** [comment]
CI shouls pass now. :) PTAL.

**Comment by steveeJ** [comment]
This is a good opportunity to work towards a style guide, which we haven't defined yet for the Enarx organization. 

I'm in favor of named arguments and fields while I see these two possible cons: they require public fields, and they are prone to breaking compatibility (which might not be a con but rather intention). The latter is easily solvable by deriving `Default` and using the `..Default::default()` notation at the end of each instantiation. The former is a limitation in Rust itself, which I haven't investigated in depth.

Since the structs we're dealing with here already uses `pub` fields, we could think about introducing the `Default` approach here, depending on whether we can foresee adding new fields which we expect to have defaults. In this case these could be patch and build version fields.

**Comment by zeenix** [comment]
Thanks for your input on this @steveeJ .

> depending on whether we can foresee adding new fields which we expect to have defaults

Although I'm a complete newbie to Enarx, I don't foresee addition of fields of any non-simple external type w/o a Default implementation in the future to these simple structs. But if that prediction turns out to be wrong, we can always create a wrapper type, implement Default for it and use that instead.

So I think I'll just add Default implementation here and it should be good for now. Unless @npmccallum or anyone disagrees?

**Comment by zeenix** [comment]
@steveeJ Turns `Default` was already implemented through `derive` on both structs so I've now modified the patch to make use of it where it makes sense. PTALA.

File: src/certs/sev/cert/v1/body/mod.rs [hunk 1]
```diff
@@ -39,7 +39,7 @@ impl Body {
             Body {
                 ver: 1u32.to_le(),
                 data: Data {
-                    firmware: crate::Version(0, 0),
+                    firmware: Default::default(),
                     reserved: 0,
                     key,
                 },
```
Comment (npmccallum) [inline src/certs/sev/cert/v1/body/mod.rs]: Is initialization of this struct eligible for `Default::default()`? If not, I think this patch looks good.

Comment (zeenix) [reply src/certs/sev/cert/v1/body/mod.rs]: You mean the `Data` struct implementing `Default`? The `key` field there makes it a bit complicated. That means we need to implement `Default` for a bunch of nested structs. Not saying whether it's a good idea or not, just that it'd be a separate change if it's what we want.

Comment (npmccallum) [reply src/certs/sev/cert/v1/body/mod.rs]: I think that answers my question in the negative. LGTM

**Pull request merged**